lib/src/model/puzzle/streak_storage.dart

```diff
@@ -12,12 +13,24 @@ part 'streak_storage.g.dart';
 
 @Riverpod(keepAlive: true)
 StreakStorage streakStorage(Ref ref, UserId? userId) {
-  return StreakStorage(userId);
+  return StreakStorage(ref, userId);
+}
+
+/// Fetches the current streak score from the local storage if available, returns null otherwise.
+@riverpod
+Future<int?> savedStreakScore(Ref ref) async {
+  final session = ref.watch(authSessionProvider);
+  // cannot use ref.watch because it would create a circular dependency
+  // as we invalidate this provider in the storage saveActiveStreak and clearActiveStreak methods
+  final streakStorage = ref.read(streakStorageProvider(session?.user.id));
```

Comment:
The proper way to deal with this imo, is to invalidate the `streakStorageProvider` directly (better to self invalidate than invalidate another provider to avoid such issues).

Reply:
Right, FYI I copied this from https://github.com/lichess-org/mobile/blob/a6a8c0e719cdf4069cedb68208e7a28501f10563/lib/src/model/correspondence/correspondence_game_storage.dart#L26, so maybe it should be changed over there as well

Reply:
Hmm but if I change it to `ref.invalidate(streakStorageProvider);` it breaks the `StreakScreen`:

[streakBroken.webm](https://github.com/user-attachments/assets/ade1aa23-256b-4016-84cd-46afb4efc30e)

Reply:
If `clearActiveStreak` invalidates the whole provider, the `GAME OVER` text is missing and we go right to the next puzzle.

That's probably also the reason why it's not done that way for offline correspondence?

Reply:
> hmm actually I noticed some other issues with streak which I'm fixing now and invalidating storage directly is not really possible because it makes reload other providers that we don't want to. So I'll leave it as you did.

Alright! Sorry I didn't see this comment before I wrote the two comments above